### DISCUSSION TITLE
Wasm-wasi support

### DISCUSSION THREAD

**Discussion by Tails** [rust-skia/rust-skia#711] [General]
I'm trying to draw a server-side canvas inside a [WasmEdge WASM container](https://crates.io/crates/wasmedge_quickjs), but it fails with:

```bash
error: failed to run custom build command for `skia-bindings v0.50.0`

Caused by:
  process didn't exit successfully: `target/release/build/skia-bindings-55f8bc266dfe1faf/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-env-changed=SKIA_DEBUG

  --- stderr
  thread 'main' panicked at 'Failed to parse TARGET wasm32-wasi', .cargo/registry/src/github.com-1ecc6299db9ec823/skia-bindings-0.50.0/build_support/cargo.rs:129:5
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
make: *** [build-bin] Error 101
``` 

I'm guessing the environment parser expects 3 parts like with wasm32-unknown-unknown.

**Comment by pragmatrix** [comment]
There is only target support for wasm32-unknown-emscripten so far, and I have no idea what needs to be done to port or build Skia and skia-bindings to other wasm targets. So I'd suggest you might dig a deeper rabbit hole, or look for more portable alternatives like tiny-skia.
